### PR TITLE
Add CreateICW to Interop Gdi32

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Add CreateICW to Interop Gdi32.
- Remove CreateIC from UnsafeNativeMethods and replace its usages.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/dotnet/winforms/pull/2881)

### CODE REVIEW

**Comment by gpetrou** [comment]
I thought that when we just pass parameters it is OK to use string? Also, looking at the code coverage seems like the RightMargin property lines are already covered?

**Comment by RussKie** [comment]
> I thought that when we just pass parameters it is OK to use string?

TL;DR: it works, but it is not efficient.

`string` isn't [blittable](https://docs.microsoft.com/en-us/dotnet/framework/interop/blittable-and-non-blittable-types), and that means another string will need to be allocated in the unmanaged heap. By making the signature `char*` we pin the managed string and let the unmanaged code to access it directly.
This is the overhead I mentioned earlier.



> RightMargin property lines are already covered?

There is a single test that asserts `RightMargin == 0`.
We need tests that set values with and without `Handle` created, and assert it works correctly.

**Comment by gpetrou** [comment]
In RichTextBoxTests I see many calls to RightMargin, not one. Am I missing something here?

**Comment by RussKie** [comment]
> In RichTextBoxTests I see many calls to RightMargin, not one. Am I missing something here?

You're right! I was looking at an outdated branch (I have several working repos).
So we should be in a good position to refactor it then.

File: src/System.Windows.Forms.Primitives/src/Interop/Gdi32/Interop.CreateICW.cs [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+        [DllImport(Libraries.Gdi32, ExactSpelling = true, SetLastError = true, CharSet = CharSet.Unicode)]
+        private static extern unsafe IntPtr CreateICW(char* lpszDriverName, char* lpszDeviceName, char* lpszOutput, IntPtr /*DEVMODE*/ lpInitData);
+
+        public static unsafe IntPtr CreateICW(string lpszDriverName, string lpszDeviceName, string lpszOutput, IntPtr /*DEVMODE*/ lpInitData)
+        {
+            fixed (char* driverName = lpszDriverName)
+            {
+                fixed (char* deviceName = lpszDeviceName)
+                {
+                    fixed (char* output = lpszOutput)
```
Comment (RussKie) [inline src/System.Windows.Forms.Primitives/src/Interop/Gdi32/Interop.CreateICW.cs]: [nit] in cases like that it is ok to omit the outer braces to make the code slightly more compact:
```cs
fixed (char* driverName = lpszDriverName)
fixed (char* deviceName = lpszDeviceName)
fixed (char* output = lpszOutput)
{
}
```
https://github.com/dotnet/winforms/blob/9dc2a1d01e1f7b2383acd232665307d9adec50b3/src/System.Windows.Forms.Primitives/src/Interop/ComCtl32/Interop.TTOOLINFOW.cs#L58-L63

**Pull request merged**